jwst/resample/resample_spec_step.py

```diff
@@ -25,6 +25,23 @@ class ResampleSpecStep(ResampleStep):
     def process(self, input):
         input = datamodels.open(input)
 
+        if isinstance(input, ImageModel):
+            slit_model = datamodels.SlitDataModel()
+            input.meta.bunit_err = None  # this prevents error array populated in slit model
+            # copy the meta data from input do slit_model
+            slit_model.update(input)
+            slit_model.update(input,only='PRIMARY') # tried this it does not help populate Primary meta
+            slit_model.data = input.data
+            slit_model.meta  = input.meta
```

Comment:
Why is this necessary, if .update() has been called? Shouldn't .update() have already copied over all the meta?

Reply:
Ok changing the data model to SlitModel - fixed the problem with the meta data not being copied over.

Reply:
Also, the statement `slit_model.meta  = input.meta` should not be done. This is making the two models share the meta structure. This is what the `update` should be handling.

Reply:
Looking a bit deeper, this section of code should look like:
```
            slit_model = datamodels.SlitModel()
            slit_model.update(input)
            slit_model.meta.wcs = input.meta.wcs
            slit_model.data = input.data
            input = slit_model
```

The next issue is that in `_process_slit`, the call `resamp.do_drizzle` is returning a result without the meta being properly updated. Not sure when that is supposed to occur, whether resample is supposed to take care of it or whether the callee is supposed to.